### ISSUE TITLE
Issue with treatment identification (ID vs. characters)

### ISSUE THREAD

**Issue by CatherineRoyer** [gertvv/gemtc#64]
I have performed an NMA (risk ratio) with ID as treatment (numbers) vs. same NMA with characters (treatment names) as treatment; and results differ. I have performed the NMA in frequentist model to verify and results are in line with results using tx ID. I think it might have an issue when using treatment as characters and not as ID (numbers).

**Comment by CatherineRoyer** [comment]
for example, one node (RR 95% CrI) = 0.8 (0.2; 2.9) with Tx ID vs. 1.8 (0.5; 7.0) with Tx as characters.

**Comment by gertvv** [comment]
What version of R and gemtc are you using?

**Comment by CatherineRoyer** [comment]
the last version. I also tried with a previous version. same issue

**Comment by CatherineRoyer** [comment]
gemtc_1.0-0

**Comment by CatherineRoyer** [comment]
Please see attached, I replicated the issue with fake treatment name, thanks,
[data.xlsx](https://github.com/gertvv/gemtc/files/6469746/data.xlsx)
[Code.txt](https://github.com/gertvv/gemtc/files/6469747/Code.txt)

**Comment by gertvv** [comment]
Ah, I see. As you have defined your treatments data frame with numeric IDs, you should refer to them by their numeric IDs in the data. I do think gemtc is being a bit too permissive here though, I'll see if I can protect against incompatible treatment IDs between the arguments. Thanks for reporting.

**Comment by gertvv** [comment]
No, my mistake - looks like something's up with relative.effect.table

**Comment by gertvv** [comment]
Actually the problem occurs earlier. Even though I don't see anything obviously wrong with how the models are being formulated, the results are indeed different. I think this is not because of the treatment names, but because the "reference treatment" in the name-based model doesn't have index 1, whereas in the id-based model it does. If I use treatment IDs, but assign them in the same order the name-based model does, I get the same results:

```
map <- match(network$treatments$description, network_txName$treatments$description)
dataNMA_id2 <- dataNMA_id
dataNMA_id2$treatment <- map[dataNMA_id2$treatment]
data_treatments2 <- data.frame(id=1:13, description=as.character(network_txName$treatments$description))
network_id2 <- mtc.network(data.ab=dataNMA_id2, treatments=data_treatments2)
model_id2 <- mtc.model(network_id2, likelihood = "binom", link = "log",  linearModel = "fixed")
results_id2 <- mtc.run(model_id2, n.adapt =5000, n.iter = 20000)
```

**Comment by CatherineRoyer** [comment]
Yes the problem occurs earlier. You are right, it's about the reference treatment (number 1). I tried without an ID Table, and with placebo named "1_Placebo" and other treatments named by their name and not ID (without numbers). Results = the same as with tx ID. (see data from "Data1(ref)")


network_ref <- mtc.network(dataNMA_ref)


model_ref  <- mtc.model(network_ref , likelihood = "binom", link = "log",  linearModel = "fixed")
results_ref <- mtc.run(model_ref , n.adapt =50000, n.iter = 20000)
[data.xlsx](https://github.com/gertvv/gemtc/files/6472033/data.xlsx)

rel.effect.table_ref <- relative.effect.table(results_ref)
gemtc::forest(rel.effect.table_ref , "Catherine")

**Comment by gertvv** [comment]
Thanks Catherine. I reduced this to a 3-treatment dataset (Placebo, Catherine, Anne) and still was different results. The problem occurs on the log-Risk Ratio (binom/log) scale but not on the log-Odds Ratio (binom/logit) scale.

Using the Risk Ratio for network meta-analysis in general is not ideal, I always recommend to use the Odds Ratio for the analysis, and then calculate Risk Ratios using an assumed (or empirical!) risk or risk distribution for one of the treatments.

This example seems a bit extreme though, I'll try to dig into it a bit more.

**Comment by gertvv** [comment]
It might be the prior for the study reference arms, as setting a different treatment order also reorders the arms in the dataset, causing the prior to affect a different arm.

Risk Ratio:
```
	  mu[i] <- log(p.base[i])
	  p.base[i] ~ dunif(0, 1)
```

Odds Ratio:
```
	  mu[i] ~ dnorm(0, prior.prec)
```

**Comment by gertvv** [comment]
The priors are indeed the cause. I've changed the prior to:
```
  mu[i] ~ dnorm(log(0.5), prior.prec) T(,0)
```
As the log risk has its neutral value at log(0.5) and an upper bound at log(1) = 0.

The results of the two models now agree (note the first 4 treatments are re-ordered between the plots):

![Rplot002](https://user-images.githubusercontent.com/80785/118126561-cb5c0980-b3f8-11eb-91a8-0f583ab446e2.png)
![Rplot001](https://user-images.githubusercontent.com/80785/118126564-cbf4a000-b3f8-11eb-8281-214913464919.png)

**Comment by CatherineRoyer** [comment]
what should I do now ? is the problem resolved in the package ? and should I use a treatment table ?

**Comment by CatherineRoyer** [comment]
on another note, I tried to change the order in the treatment table, and it seems the id coluum is not used. instead, just the order of the treatment description that is used 
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/croyer/AppData/Local/Temp/msohtmlclip1/04/clip.htm">
<link rel=File-List
href="file:///C:/Users/croyer/AppData/Local/Temp/msohtmlclip1/04/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



id | description | order
-- | -- | --
9 | Placebo | 1
10 | Catherine | 2
11 | Annie | 3
12 | Anne | 4
13 | Beta | 5
1 | Ile | 6
2 | Lune | 7
3 | Lupa | 8
4 | Olivier | 9
5 | Patte | 10
6 | Quebec | 11
7 | Rocher | 12
8 | Zizaz | 13



</body>

</html>

**Comment by CatherineRoyer** [comment]
rel.effect.table <- relative.effect.table(results)
gemtc::forest(rel.effect.table, 2, use.description = TRUE)

![image](https://user-images.githubusercontent.com/84083157/118128930-88367200-b3c9-11eb-95b7-4483c98b24cb.png)


you see the forest plot used "Catherine", the second tx instead of the id #2 (Lune) as in the formula.

**Comment by CatherineRoyer** [comment]
is it possible to know when will be the next update of the package with this correction ?

**Comment by gertvv** [comment]
I'm hoping it'll be in the next few days, need to make sure all the tests pass on different versions of R, then submit to CRAN.

**Comment by CatherineRoyer** [comment]
Thank you very much for this fix, very appreciated !

**Comment by gertvv** [comment]
The fix is on CRAN now (version 1.0-1).

**Comment by CatherineRoyer** [comment]
Thanks Gert !
